src/Domain/Money.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace MatchBot\Domain;
+
+use MatchBot\Application\Assertion;
+
+/**
+ * @psalm-immutable
+ */
+class Money
```

Comment:
We could use a third party library like https://github.com/brick/money or https://github.com/moneyphp/money to save us some work, either instead of this class or inside of it. Might very well be worth it before we make this class more complicated, but as for now we have just one very basic use case it seemed simpler to write this class than make a decision about what third partly library to use if any.